### PR TITLE
refactor(executor): rewrite whole imap executor to add new commands

### PR DESCRIPTION
**Pull request by tomadele**
New commands: append, create, clear, delete, fetch, flag, move

### CODE REVIEW

**Comment by tomadele** [comment]
Having troubles with signed commits, will check later.

File: executors/imap/README.md [hunk 1]
```diff
@@ -1,51 +1,274 @@
...
+- **flag**: add, remove or set the flags of a mail
+- **move**: move a mail from one mailbox to another
+
+Multiple commands can be stated as once: they will be executed in the order they are presented under the `commands` field.
+
+Each command consists of a `name` field which allows to specify the command to execute.
+Then they are composed of the `search` field and/or the `args` field.
+
+The `search` field always has the same syntax. It retrieves and returns the first mail found through the search criteria:
+````yaml
```
Comment (benoitmasson) [inline executors/imap/README.md:35]:
```yaml

File: executors/imap/README.md [hunk 2]
```diff
@@ -1,51 +1,274 @@
...
+search:
+  mailbox: testBox # The mailbox in which to search the mail
+  uid: 1 # The UID to search for
+  from: .*@your-domain.localhost # The "From" header field to search for
+  to: you@company.tld # The "To" header field to search for
+  subject: Title of mail with * # The "Subject" header field to search for
+  body: .*a body content.* # The "Body" field to search for
+````
```
Comment (benoitmasson) [inline executors/imap/README.md:43]: "```"

File: executors/imap/README.md [hunk 3]
```diff
@@ -1,51 +1,274 @@
...
+      to: recipient-of@my-new-mail.com # OPTIONAL: the "To" header field of the new mail
+      subject: Subject of my new mail # OPTIONAL: the "Subject" header field of the new mail
+      body: Body of my new mail # OPTIONAL: the body of the new mail
+      flags: # OPTIONAL: the flags of the new mail (note that the "\Recent" flag will most probably be added as well)
+        - Flag1
+        - Flag2
+        ...
+```
+
+The result of the append command
```
Comment (benoitmasson) [inline executors/imap/README.md:70]: missing end of sentence? of result sample?

File: tests/imap.yml [hunk 4]
```diff
@@ -1,29 +1,183 @@
 name: SMTP & IMAP testsuite
```
Comment (benoitmasson) [inline tests/imap.yml:1]:
```suggestion
name: IMAP testsuite
```

Comment (benoitmasson) [inline tests/imap.yml:1]:
```suggestion
name: IMAP testsuite
```

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#1058.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#1059.0 ✘  
* Build
  * Build ✔ 
  * Unit Tests ✔ 
* Tests
  * Acceptance Tests ✘

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Comment by yesnault** [comment]
LGTM, thank you @tomadele

**Pull request merged**